R/utils.R

```diff
@@ -166,4 +166,20 @@ capitalize <- function(x) {
     paste0(toupper(substr(x, 1, 1)), substring(x, 2))
 }
 
+activeEncoding <- function(project = getwd()) {
+  defaultEncoding <- getOption("encoding")
 
+  # attempt to locate .Rproj file
+  files <- list.files(project, full.names = TRUE)
+  rprojFile <- grep("\\.Rproj$", files, value = TRUE)
+  if (length(rprojFile) != 1)
+    return(defaultEncoding)
+
+  # read the file
+  contents <- readLines(rprojFile, warn = FALSE, encoding = "UTF-8")
```

Comment:
RStudio writes some interleaving blank lines between fields, which causes a bit of confusion for `read.dcf`; e.g.:

```
> read.dcf("rstudio.Rproj", all = TRUE)
  Version RestoreWorkspace SaveWorkspace AlwaysSaveHistory EnableCodeIndexing
1     1.0             <NA>          <NA>              <NA>               <NA>
2    <NA>               No            No                No               <NA>
3    <NA>             <NA>          <NA>              <NA>                Yes
4    <NA>             <NA>          <NA>              <NA>               <NA>
  UseSpacesForTab NumSpacesForTab Encoding RnwWeave    LaTeX
1            <NA>            <NA>     <NA>     <NA>     <NA>
2            <NA>            <NA>     <NA>     <NA>     <NA>
3             Yes               3    UTF-8     <NA>     <NA>
4            <NA>            <NA>     <NA>   Sweave pdfLaTeX
```